### ISSUE TITLE
400 codellama:latest does not support tools

### ISSUE THREAD

**Issue by masterwishx** [clinebot/cline#460]
### Which API Provider are you using?

Ollama

### Which Model are you using?

codelllama

### What happened?

got error:  

`400 codellama:latest does not support tools`

### Steps to reproduce

1. install extension 
2.connect to ollama model
3.

**Comment by masterwishx** [comment]
OK founded other ollama lang that dont have this error : like `hhao/qwen2.5-coder-tools:latest`  etc

**Comment by atljoseph** [comment]
It has to do with the model file template. That error comes directly from Ollama I think. I took the qwen2.5-coder (very similar template to the tools one) and tried with default template. Worked fine although it easily got confused - par for the Ollama course. But then I removed the template and then this EXACT error was reported. I put the template back, and it went away. That should be enough to tell that this error has nothing absolutely nothing to do with Claude dev, and everything to do with the template on which the model file is built.
